### ISSUE TITLE
tags.tiqcdn.com is blocked, but is not strictly "advertisement" related

### ISSUE THREAD

**Issue by LordLingham** [StevenBlack/hosts#1103]
tags.tiqcdn.com is blocked which prevents Tealium's utag.js file from loading on webpages.

Tealium is only a "Tag Manager", which means while utag.js facilitates the loading of other tags / advertisements / tracking scripts, the files themselves loaded from this domain are not strictly tracking the users.

Some websites which use utag.js rely on adding additional scripts to the page to provide other features to their websites such as error tracking, live chats and consent manager.
Blocking utag.js from loading, and other files from this domain removes functionality from the site, and potentially "breaks" their site for user.

Is it possible to whitelist / remove the block for this domain, as the URL's which would be loaded from utag for the advertising / analytics vendors are likely already covered within the block list.

**Comment by spirillen** [comment]
As you clearly (to me) write it yourself

> `utag.js` facilitates the loading of other tags / advertisements / tracking scripts

If blocking of this breaks your site, they are clearly miss-written and needs a heavy rewrite of the code.

Blocking badware that breaks a site is not something I can see is our problem; It yours, as your intention clearly is to violate the freedom of the internet. By which it was developed in 1988.

That's my opinion, beside of that, you would have to take it up with those how maintain the list(s), from which this occurs.  In this case it is among others [mvps.org](#records)

@StevenBlack's list is only a collection of different hosts files.

And for your information, then I'm one of them blocking your spyware, unwanted-ware

```
tags.tiqcdn.com.rpz.mypdns.cloud
akamai.tags.tiqcdn.com.rpz.mypdns.cloud
edgecast.tags.tiqcdn.com.rpz.mypdns.cloud
```

## Records
```
grep -Ri 'tags.tiqcdn.com' data/
data/mvps.org/hosts:# 0.0.0.0 tags.tiqcdn.com #[affects Intel.com/Shaw]
```

:smiley: :rofl:

**Comment by StevenBlack** [comment]
@spirillen please be nice to the visitors, here.  Thank you.

**Comment by StevenBlack** [comment]
James @LordLingham the `tags.tiqcdn.com` domain comes to us via [mvps.org](http://winhelp2002.mvps.org/) who run a very good operation.  You will need to make your case with there.

Therefore please take this issue to them at winhelp2002@gmail.com.  They'll do the right thing.

**Comment by LordLingham** [comment]
Hi @StevenBlack Thanks for the response.

I had a report of issues with pihole, a quick look suggested this was one of the lists used by default with pihole, and as far as I can tell, the only list in the default for pihole which contained this domain.

If this is commented out by default, It feels safe to assume the user has uncommented this / added the domain in manually, implying they DO want this to be blocked (no problems there)
The main concern was with this being included and uncommented by default, causing this to always be blocked.

@spirillen  - While utag.js can be used to load these advertising scripts, it also can provide other functionality to the site, which is lost when the script is not loaded.
This was not me trying to circumvent blocking of tracking / advertising scripts, just trying to ensure a more targeted approach, relying on blocking the specific URL's rather than the "middle man".

I'll drop an email to mvps org and see how things go.
Thanks for the input from both of you.

**Comment by spirillen** [comment]
Hi @LordLingham as you right your self

> While utag.js _can_ be used to load these advertising scripts

=> And therefor will be and does.

How should a endures be able to now when it's running as evil as possible or not?

That's what tricked my first response and why it is blocked.

So in case your issue is about you are using this "evil" code on one of your sites my personal recommendation is to replace it with a non evil code.

**Comment by LordLingham** [comment]
@spirillen 

> How should a endures be able to now when it's running as evil as possible or not?

As mentioned utag.js acts as a "middle man" and does not host any of the "evil" scripts, but instead "loads" them from the same destination you would expect.

Google analytics or a Facebook pixel loading from Utag will still be loading from the same address as if they were not loaded through utag, which would almost certainly mean they are already blocked.

In this instance blocking utag is unnecessary as it's doubling down on preventing a tag / tracking script loading which is already blocked, while preventing the "non evil" functionality provided by the script (utag).

If these scripts were circumventing the "blocking" in a way I could understand the purpose of blocking, but utag.js is simply the middleman

**Comment by StevenBlack** [comment]
James @LordLingham, by that logic, adware and malware purveyors would use a proxy, which cloaks their origin, and somehow this is ok?  How is that ok?  How is this acceptable?

Greenwashing sources, being a "middleman" as you say, is not OK.

This isn't going to change.

**Comment by spirillen** [comment]
> James @LordLingham, by that logic, adware and malware purveyors would use a proxy, which cloaks their origin, and somehow this is ok? How is that ok? How is this acceptable?
> This isn't going to change.

Second this by all my hard :santa: 

Obfuscating potentially evil trackers/spyware whatever you would call it though a third party script, is not OK, and actually only to become a even bigger issue with the release of [APOMAYA AdGain](https://gitlab.com/my-privacy-dns/matrix/matrix/issues/1444), which will challenge many spyware/tracking fighter for the time to come by hours and hours of free work.

**Comment by LordLingham** [comment]
@StevenBlack @spirillen 
Thanks for both of your inputs on this.

I 100% agree with you guys here about being against trying to "greenwash" things and try and sneak past things past blockers like this. If utag.js was doing things like this, I wouldn't be advocating it's removal from a black list like this.

I'm worried that based off these replies. I either haven't been clear about what utag.js does, or I'm misunderstanding your responses.

It sounds like I've given the impression that utag.js "masks" these scripts and hides what is being loaded and gives a false sense of "not being tracked".
Utag.js does not "interfere" with the script loading, and makes no attempt to hide / justify loading any scripts it simply loads the scripts it is told.

Loading a Facebook pixel through utag.js is no different to "hard coding" it on your page. the network request will look identical, and would still be subject to the same identification / blocking.

Tags / scripts are not loaded through utag to evade security or "hide", but to allow easier control over what circumstances different tags fire and being able to add / remove / modify these without needing developers to update the live website for every adjustment.

Ultimately if the domain for the FB pixel is blocked by a blacklist like this, the tag isn't going to load whether it's load is initiated by utag.js or by something else on the page natively.

Steps to load a tag would be a little like this

1) Utag.js loads
2) Utag checks conditions on loading FB Pixel
3 ) utag tries to load FB PIxel
4) If FB pixel server is blocked it won't load, if unblocked it will.

Think of it as a "library" loaded on the page that serves multiple functions. Some of which will ATTEMPT to load tracking scripts such as Google analytics / FB Pixels.

If you guys have understood my point here - i won't repeat myself, to avoid us going around in circles! I just wanted to be clear there was no confusion about the role utag.js plays here.

Thanks for your time / input again.

**Comment by J0rnT** [comment]
The BUPA healthcare app now requires this URL. Frustrating for all.

**Comment by UnconnectedBedna** [comment]
Necro bump, but this was the only hit I got when searching for tiqcdn.com

My bank (nordea) _sometimes_ refuses to let me login if not allowing tags.tiqcdn.com on my pihole.
This happens sporadically, not always. Sometimes it works, sometimes it just throws a "something went wrong".
But by whitelisting tags.tigcdn.com it never fails.
Could probably be good for you guys to know...

**Comment by fipp** [comment]
Chiming in, "send message" silently fails for insurance claims Norwegian insurance company "Tryg" due to this block.

**Comment by kalirob99** [comment]
Seems that the Paramount+ app now requires this as well, it stopped working until I unblocked it. 😣
